File: c/1768-merge-strings-alternately.c
#include <stdlib.h>

char* mergeAlternately(const char* word1, const char* word2) {
    int len1 = strlen(word1), len2 = strlen(word2);
    int minLen = len1 < len2 ? len1 : len2;
    int totalLen = len1 + len2;
    char* result = (char*)malloc(totalLen + 1);
    if (!result) return NULL; 

    int i, j = 0;
    for (i = 0; i < minLen; i++) {
        result[j++] = word1[i];
        result[j++] = word2[i];
    }
    while (i < len1) result[j++] = word1[i++];
    while (i < len2) result[j++] = word2[i++];

    result[j] = '\0';
    return result;
}

File: go/1768-merge-strings-alternately.go
package main

func MergeAlternately(word1 string, word2 string) string {
	result := make([]byte, 0, len(word1)+len(word2))
	minLen := len(word1)
	if len(word2) < minLen {
		minLen = len(word2)
	}

	for i := 0; i < minLen; i++ {
		result = append(result, word1[i], word2[i])
	}
	result = append(result, []byte(word1[minLen:])...)
	result = append(result, []byte(word2[minLen:])...)
	return string(result)
}
